### ISSUE TITLE
--query matches variants furthest away from queried insertion in cases where ins_distance > 0

### ISSUE THREAD

**Issue by alkc** [J35P312/SVDB#77]
Hej, 

I'm working with updating a really old version of SVDB in an analysis pipeline to version 2.8.2 and am trying to understand some of the diffs in variant annotations that I see between the two versions, which seem to mostly concern INS-type variants. I've familiarized myself with the new `--ins_distance` param (yes, _that old_ version of SVDB) to both `--merge` and `--query` and have been poking around in the source code trying to understand how SVDB matches insertions in the query module. 

The text that follows assumes `use_OCC_tag = True`. From what I gather, the matches are made in the block below, where in cases there is a match, `hit_tmp` will be the absolute distance between the query and matching database variant. The distance only concerns the `POS` of the query and database variant and not end or length:
https://github.com/J35P312/SVDB/blob/3e95936b13dda6507d4f9648824727068b4174ea/svdb/query_module.py#L226-L229

This distance and the database variant's count and frequency are appended to `similarity`, `occ` and `freq` respectively here:

https://github.com/J35P312/SVDB/blob/3e95936b13dda6507d4f9648824727068b4174ea/svdb/query_module.py#L234-L238

From this list of variants a selection is then made here on L248-249, where the index `idx` to `occ` and `freq` is selected by finding `max(similarity)`. But since `similarity` in this case is an array of absolute distances then the selected db variant will be the one with the furthest distance to the queried insertion, for cases where `--ins_distance >0`  and the matching variants contain at least one variant with a returned distance is larger than that of the other matches. This means that the matches will get worse for larger values of `--ins_distance`:

https://github.com/J35P312/SVDB/blob/3e95936b13dda6507d4f9648824727068b4174ea/svdb/query_module.py#L244-L252

I've created a branch here where I've enabled some debug loggers that prints the intermediate steps in the matching: https://github.com/alkc/SVDB/tree/alkc/debug-logging

For an ins_distance of `1000` (just to illustrate): the output after the accumulation of matches against one queried variant will look like this for a variant in my case (formatted a bit for readability):

```
" vector of occ for search hits:
DEBUG:svdb.query_module:occ=[
np.str_('3'), np.str_('23'), np.str_('1'), np.str_('1'), np.str_('2'), np.str_('1'), np.str_('5'), np.str_('13'), np.str_('6'),  
np.str_('2'), np.str_('3821'), np.str_('1'), np.str_('1'), np.str_('1'), np.str_('5'), np.str_('14'), np.str_('33'), np.str_('4'), np.str_('4')
]              

# corresponding similarity vector for search hits, ranges from 0 to 43:
DEBUG:svdb.query_module:similarity=[
np.int64(43), np.int64(10), np.int64(10), np.int64(8), np.int64(5), np.int64(4), np.int64(3), np.int64(2), np.int64(0), np.int64(0), np.int64(0), np.int64(0), np.int64(1), np.int64(2), np.int64(3), np.int64(4), np.int64(4), np.int64(10), np.int64(10)
]                            

# selection of idx by max similarity:
DEBUG:svdb.query_module:similarity[0]=43 # <- This is the selected similarity, with `idx = 0` and `dist = 43`

DEBUG:svdb.query_module:from occ=[
np.str_('3'), np.str_('23'), np.str_('1'), np.str_('1'), np.str_('2'), np.str_('1'), np.str_('5'), np.str_('13'), np.str_('6'), np.str_('2'), np.str_('3821'), np.str_('1'), np.str_('1'), np.str_('1'), np.str_('5'), np.str_('14'), np.str_('33'), np.str_('4'), np.str_('4')
] 

and 

from freq=[
np.str_('0.00059'), np.str_('0.00451'), np.str_('0.0002'), np.str_('0.0002'), np.str_('0.00039'), np.str_('0.0002'), np.str_('0.00098'), np.str_('0.00255' ), np.str_('0.00118'), np.str_('0.00039'), np.str_('0.7498'), np.str_('0.0002'), np.str_('0.0002'), np.str_('0.0002'), np.str_('0.00098'), np.str_('0.00275'), np.str_('0.00648'), np.str_('0.00078'), np.str_('0.00078')

], select occ[0]=3,  frequency[0]=0.00059                                                          
```

in short the variant 43 bases away from the queried variant is picked over, say, `idx=10` with `dist=0` and `occ=3821` which would've been the interesting one to return in this case (that one won't be returned for `ins_distance=0 `either I've found, but that's something I need to look into more).
